src/pybind/rados.py

```diff
@@ -1821,16 +1847,204 @@ def lookup_snap(self, snap_name):
+           :type int
+           return:object(OmapIterator),return value from this action
+        """
+        prval = c_int()
+        iter_addr = c_void_p()
+        run_in_thread(self.librados.rados_read_op_omap_get_vals,
+                      (c_void_p(read_op), c_char_p(start_after),
+                       c_char_p(filter_prefix), c_int(max_return),
+                       byref(iter_addr), pointer(prval)))
+        return OmapIterator(self, iter_addr), prval.value
+
+    @requires(('read_op', int), ('start_after', str), ('max_return', int))
+    def get_omap_keys(self, read_op, start_after, max_return):
+        """
+           get the omap keys
+           :para read_op :read operation object
+           :type int
+           :para start_after : list keys starting after start_after
+           :type str
+           :para max_return:list no more than max_return key/value pairs
+           :type int
+           return:object(OmapIterator),return value from this action
+        """
+        prval = c_int()
+        iter_addr = c_void_p()
+        run_in_thread(self.librados.rados_read_op_omap_get_keys,
+                      (c_void_p(read_op), c_char_p(start_after),
+                       c_int(max_return), byref(iter_addr), pointer(prval)))
+        return OmapIterator(self, iter_addr), prval.value
+
+    @requires(('read_op', int), ('keys', tuple))
+    def get_omap_vals_by_keys(self, read_op, keys):
+        """
+           get the omap values by keys
+           :para read_op :read operation object
+           :type int
+           :para keys : input key tuple
+           :type tuple
+           return:object(OmapIterator),return value from this action
+        """
+        prval = c_int()
+        iter_addr = c_void_p()
+        key_num = len(keys)
+        key_array_type = c_char_p*key_num
+        key_array = key_array_type()
+        key_array[:] = keys
+        run_in_thread(self.librados.rados_read_op_omap_get_vals_by_keys,
+                      (c_void_p(read_op), byref(key_array), c_int(key_num),
+                       byref(iter_addr), pointer(prval)))
+        return OmapIterator(self, iter_addr), prval.value
+
+    @requires(('write_op', int), ('keys', tuple))
+    def remove_omap_keys(self, write_op, keys):
+        """
+           remove omap keys specifiled
+           :para write_op :write operation object
+           :type int
+           :para keys : input key tuple
+           :type tuple
+        """
+        key_num = len(keys)
+        key_array_type = c_char_p*key_num
+        key_array = key_array_type()
+        key_array[:] = keys
+        run_in_thread(self.librados.rados_write_op_omap_rm_keys,
+                      (c_void_p(write_op), byref(key_array), c_int(key_num)))
+    
+    @requires(('write_op', int))
+    def clear_omap(self, write_op):
+        """
+           Remove all key/value pairs from an object
+           :para write_op :write operation object
```

Comment:
i am not sure if the ending space would help. please see http://sphinx-doc.org/domains.html#info-field-lists for more details on how we document the python code.